### PR TITLE
feature(partner_sdk): fix java examples as per updated proto files

### PR DESCRIPTION
**Pull request by fivetran-niketkhandelwal**
Closes https://fivetran.height.app/T-676968, https://fivetran.height.app/T-676972

Testing details:

1. tested connector example changes using connector tester
Screenshot 2024-05-07 at 20 52 54
2. tested destination example changes using destination tester
Screenshot 2024-05-07 at 20 57 15

### CODE REVIEW

File: destination_sdk.proto [hunk 1]
```diff
@@ -73,10 +73,10 @@ message AlterTableRequest {
     map<string, string> configuration = 1;
     string schema_name = 2;
     string table_name = 3;
-    repeated SchemaChange changes = 4;
+    repeated DestinationSchemaChange changes = 4;
```
Comment (ediril) [inline destination_sdk.proto:76]: Please change this back as well

File: connector_sdk.proto [hunk 2]
```diff
@@ -67,12 +67,12 @@ message TableSelection {
 message UpdateResponse {
     oneof operation {
         Record record = 1;
-        SchemaChange schema_change = 2;
+        ConnectorSchemaChange schema_change = 2;
```
Comment (ediril) [inline connector_sdk.proto:70]: Please don't rename fields without talking to me first. Let's rename this back to what it was

**Review by ediril — CHANGES_REQUESTED** [review]
- The PR description is blank. Please put a link to the ticket you are addressing.
- I renamed the PR to be just to fix java examples

File: connector_sdk.proto (hunk 2 above)
Comment (fivetran-niketkhandelwal) [reply connector_sdk.proto]: updated

File: examples/connector/java/src/main/java/connector/ConnectorServiceImpl.java [hunk 3]
```diff
@@ -17,22 +17,22 @@ public void configurationForm(ConfigurationFormRequest request, StreamObserver<C
                         .setTableSelectionSupported(true)
                         .addAllFields(Arrays.asList(
                                 FormField.newBuilder()
-                                        .setName("apikey").setLabel("API key").setRequired(true).setTextField(TextField.PlainText).build(),
+                                        .setSingle(Field.newBuilder().setName("apiKey").setLabel("API Key").setRequired(true).setTextField(TextField.PlainText).build())
+                                        .build(),
```
Comment (ediril) [inline examples/connector/java/src/main/java/connector/ConnectorServiceImpl.java:21]: Let's also use the new `placeholder` and `default_value` fields in the example

File: examples/destination/java/src/main/java/destination/DestinationServiceImpl.java [hunk 4]
```diff
@@ -15,25 +18,25 @@ public void configurationForm(ConfigurationFormRequest request, StreamObserver<C
                         .setTableSelectionSupported(true)
                         .addAllFields(Arrays.asList(
                                 FormField.newBuilder()
-                                        .setName("host").setLabel("Host").setRequired(true).setTextField(TextField.PlainText).build(),
+                                        .setSingle(Field.newBuilder().setName("host").setLabel("Host").setRequired(true).setTextField(TextField.PlainText).build())
+                                        .build(),
```
Comment (ediril) [inline examples/destination/java/src/main/java/destination/DestinationServiceImpl.java:22]: Same comment as the connector example

File: examples/destination/java/src/main/java/destination/DestinationServiceImpl.java [hunk 5]
```diff
@@ -112,4 +115,9 @@ public void writeBatch(WriteBatchRequest request, StreamObserver<WriteBatchRespo
         responseObserver.onNext(WriteBatchResponse.newBuilder().setSuccess(true).build());
         responseObserver.onCompleted();
     }
+
+    private static String getChangesFromRequest(AlterTableRequest request) {
+        final List<SchemaDiff> changes = request.getChangesList();
+        return changes.stream().map(AbstractMessage::toString).collect(Collectors.joining(", "));
```
Comment (ediril) [inline examples/destination/java/src/main/java/destination/DestinationServiceImpl.java:121]: No need to create a separate method for this

**Review by ediril — CHANGES_REQUESTED** [review]
What about `Capabilities` method?

File: examples/destination/java/src/main/java/destination/DestinationServiceImpl.java (hunk 5 above)
Comment (fivetran-niketkhandelwal) [reply examples/destination/java/src/main/java/destination/DestinationServiceImpl.java]: updated

File: examples/destination/java/src/main/java/destination/DestinationServiceImpl.java (hunk 4 above)
Comment (fivetran-niketkhandelwal) [reply examples/destination/java/src/main/java/destination/DestinationServiceImpl.java:23]: added

File: examples/connector/java/src/main/java/connector/ConnectorServiceImpl.java (hunk 3 above)
Comment (fivetran-niketkhandelwal) [reply examples/connector/java/src/main/java/connector/ConnectorServiceImpl.java:22]: added

**Comment by fivetran-niketkhandelwal** [comment]
> What about `Capabilities` method?

added example

File: examples/connector/java/src/main/java/connector/ConnectorServiceImpl.java [hunk 6]
```diff
@@ -81,101 +83,73 @@ public void schema(SchemaRequest request, StreamObserver<SchemaResponse> respons
...
 
             // -- Send a log message
-            responseBuilder.clear();
-            responseObserver.onNext(responseBuilder
-                    .setLogEntry(LogEntry.newBuilder()
-                            .setLevel(LogLevel.INFO)
-                            .setMessage("Sync STARTING")
-                            .build())
-                    .build());
+            System.out.println("[Update]: Sync STARTING");
```
Comment (manjutapali) [inline examples/connector/java/src/main/java/connector/ConnectorServiceImpl.java:96]: log should be in json format - https://github.com/fivetran/fivetran_sdk/blob/main/development-guide.md#logging

File: examples/connector/java/src/main/java/connector/ConnectorServiceImpl.java [hunk 7]
```diff
@@ -81,101 +83,73 @@ public void schema(SchemaRequest request, StreamObserver<SchemaResponse> respons
...
 
             // -- Send a log message
-            responseBuilder.clear();
-            responseObserver.onNext(responseBuilder
-                    .setLogEntry(LogEntry.newBuilder()
-                            .setLevel(LogLevel.INFO)
-                            .setMessage("Sync DONE")
-                            .build())
-                    .build());
+            System.out.println("[Update]: Sync DONE");
```
[the remark this answers is not in the document]
Comment (fivetran-niketkhandelwal) [reply examples/connector/java/src/main/java/connector/ConnectorServiceImpl.java]: updated

File: examples/connector/java/src/main/java/connector/ConnectorServiceImpl.java [hunk 8]
```diff
@@ -81,101 +87,73 @@ public void schema(SchemaRequest request, StreamObserver<SchemaResponse> respons
...
 
             // -- Send a log message
-            responseBuilder.clear();
-            responseObserver.onNext(responseBuilder
-                    .setLogEntry(LogEntry.newBuilder()
-                            .setLevel(LogLevel.INFO)
-                            .setMessage("Sync STARTING")
-                            .build())
-                    .build());
+            System.out.println(new LogEntry(INFO, "[Update]: Sync STARTING", SDK_CONNECTOR));
```
Comment (ediril) [inline examples/connector/java/src/main/java/connector/ConnectorServiceImpl.java:100]: Could we use a proper logger instead of `System.out.println()`? And then perhaps we could also eliminate the new `LogEntry` class too?

**Review by ediril — COMMENTED** [review]
Looks good. I added a comment about logging and LogEntry class

File: examples/connector/java/src/main/java/connector/ConnectorServiceImpl.java (hunk 8 above)
Comment (fivetran-niketkhandelwal) [reply examples/connector/java/src/main/java/connector/ConnectorServiceImpl.java]: updated

File: examples/connector/java/build.gradle [hunk 9]
```diff
@@ -35,6 +35,8 @@ dependencies {
 
     implementation 'com.fasterxml.jackson.core:jackson-databind:2.14.1'
     implementation "com.fasterxml.jackson.core:jackson-core:2.14.1"
+
```
Comment (ediril) [inline examples/connector/java/build.gradle:38]: why space?

File: examples/connector/java/src/main/java/connector/ConnectorServiceImpl.java [hunk 10]
```diff
@@ -2,13 +2,20 @@
...
 import io.grpc.stub.StreamObserver;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.logging.Logger;
+
+public class ConnectorServiceImpl extends SourceConnectorGrpc.SourceConnectorImplBase {
 
-import java.util.*;
+    private static final Logger LOGGER = FivetranLoggerFactory.getFivetranLogger(ConnectorServiceImpl.class, MessageOrigin.SDK_CONNECTOR);
```
Comment (ediril) [inline examples/connector/java/src/main/java/connector/ConnectorServiceImpl.java:17]: No no no we don't want to release production code to the public. Please undo this change and remove all the logging classes you added.

File: examples/destination/java/src/main/java/destination/DestinationServiceImpl.java [hunk 11]
```diff
@@ -15,32 +18,50 @@ public void configurationForm(ConfigurationFormRequest request, StreamObserver<C
...
                         .build());
 
         responseObserver.onCompleted();
     }
 
+    @Override
+    public void capabilities(CapabilitiesRequest request, StreamObserver<CapabilitiesResponse> responseObserver) {
+        responseObserver.onNext(
+                CapabilitiesResponse.newBuilder()
+                        .setSupportsHistoryMode(true)
```
Comment (ediril) [inline examples/destination/java/src/main/java/destination/DestinationServiceImpl.java:53]: let's set this to false

**Review by ediril — CHANGES_REQUESTED** [review]

File: examples/connector/java/src/main/java/connector/ConnectorServiceImpl.java (hunk 10 above)
Comment (fivetran-niketkhandelwal) [reply examples/connector/java/src/main/java/connector/ConnectorServiceImpl.java]: reverted this

File: examples/destination/java/src/main/java/destination/DestinationServiceImpl.java (hunk 11 above)
Comment (fivetran-niketkhandelwal) [reply examples/destination/java/src/main/java/destination/DestinationServiceImpl.java]: updated

File: examples/destination/java/src/main/java/destination/DestinationServiceImpl.java [hunk 12]
```diff
@@ -15,32 +18,50 @@ public void configurationForm(ConfigurationFormRequest request, StreamObserver<C
...
         responseObserver.onCompleted();
     }
 
+    @Override
+    public void capabilities(CapabilitiesRequest request, StreamObserver<CapabilitiesResponse> responseObserver) {
+        responseObserver.onNext(
+                CapabilitiesResponse.newBuilder()
+                        .setSupportsHistoryMode(false)
+                        .addAllDataTypeMappings(Lists.newArrayList(
+                                DataTypeMappingEntry.newBuilder().setFivetranType(DataType.STRING).setMapTo(DestinationType.newBuilder().setName("VARCHAR").setMapTo(DataType.STRING).build()).build(),
```
Comment (ediril) [inline examples/destination/java/src/main/java/destination/DestinationServiceImpl.java:55]: We ask the partners to map all data types, so we should be doing the same in the example

File: destination_sdk.proto [hunk 13]
```diff
@@ -18,6 +19,29 @@ service Destination {
...
+
+message DataTypeMappingEntry {
+    DataType fivetran_type = 1;
+    oneof destination_type {
+        DestinationType map_to = 2;
+        bool unsupported = 3;
+    }
+}
+
+message DestinationType {
```
Comment (ediril) [inline destination_sdk.proto:37]: This is missing the max value change that @fivetran-abdulsalam was working on. Is that coming in a separate PR?

Comment (fivetran-niketkhandelwal) [reply destination_sdk.proto:37]: yes, that will be done in a separate PR

File: examples/destination/java/src/main/java/destination/DestinationServiceImpl.java (hunk 12 above)
Comment (fivetran-niketkhandelwal) [reply examples/destination/java/src/main/java/destination/DestinationServiceImpl.java]: added mappings for all data types

File: destination_sdk.proto (hunk 13 above)
Comment (ediril) [reply destination_sdk.proto:37]: which PR is that? can you link it here? Better yet, can we merge it into this one so I can see all the proto changes in one PR?

Comment (fivetran-niketkhandelwal) [reply destination_sdk.proto:37]: That [ticket](https://fivetran.height.app/T-686139) is in progress. I've created a [subtask](https://fivetran.height.app/T-699973) under it to add max value support in examples. I'll pick the subtask once the parent PR is ready

**Review by ediril — APPROVED** [review]

File: common.proto [hunk 14]
```diff
@@ -16,14 +16,36 @@ message ConfigurationFormResponse {
 }
 
 message FormField {
+  oneof field {
```
Comment (manjutapali) [inline common.proto:19]: Why proto file changes are present? Please rebase the branch before merging.

File: examples/connector/java/src/main/java/connector/ConnectorServiceImpl.java (hunk 6 above)
Comment (manjutapali) [reply examples/connector/java/src/main/java/connector/ConnectorServiceImpl.java:96]: The log should be in json format. 

`{"level": "INFO", "message": "Sync Starting"}`

File: examples/connector/java/src/main/java/connector/ConnectorServiceImpl.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (manjutapali) [reply examples/connector/java/src/main/java/connector/ConnectorServiceImpl.java:152]: same as above

**Review by manjutapali — APPROVED** [review]
looks good.

update the logs to use JSON format.

**Review by fivetran-abdulsalam — APPROVED** [review]

File: common.proto (hunk 14 above)
Comment (fivetran-niketkhandelwal) [reply common.proto]: fixed
